snorkel/analysis/utils.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import scipy.sparse as sparse
+import torch
+
+from snorkel.types import ArrayLike
+
+
+def prob_to_pred(prob: np.ndarray) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    prob
+        A [num_datapoints, num_classes] array of probabilistic labels such that each
+        row sums to 1.
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, 1] array of predictions (integers in [1, ..., num_classes])
+    """
+    assert prob.ndim == 2
+    assert (prob <= 1).all()
+
+    pred = np.argmax(prob, axis=1) + 1
+
+    n, k = prob.shape
+    assert (pred >= 1).all()
+    assert (pred <= k).all()
+
+    return pred
+
+
+def pred_to_prob(pred: np.ndarray, num_classes: int) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    pred
+        A [num_datapoints] or [num_datapoints], 1] array of predictions
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, num_classes] array of probabilistic labels with probability
+        of 1.0 in the column corresponding to the prediction
+    """
+    pred = pred.reshape(-1, 1)
+    n = pred.shape[0]
+
+    prob = np.zeros((n, num_classes))
+
+    for idx, class_idx in enumerate(pred):
```

Comment:
Should be vectorizable no?

Reply:
I was bummed that numpy didn't have a torch.scatter like op, but behold a numpy one-liner hack: 
```return np.eye(num_classes)[pred.squeeze() - 1]```